functions/private/Test-WinUtilPackageManager.ps1

```diff
@@ -20,11 +20,17 @@ function Test-WinUtilPackageManager {
     # Install Winget if not detected
     $wingetExists = Get-Command -Name winget -ErrorAction SilentlyContinue
     if ($wingetExists) {
-        $wingetVersion = [System.Version]::Parse((winget --version).Trim('v'))
+        $wingetversionfull = (winget --version)
```

Comment:
```suggestion
        $wingetversionfull = Get-WinGetVersion
```

You can use the PowerShell Cmdlet that WinGet Provides, which of course from Microsoft themselves (Source of Cmdlet(s) is `Microsoft.WinGet.Client`), may be using Cmdlet is better than saving output of command into a variable, don't really know if that's true or not, it could be personal preference in cases like these.

anyhow, good commit @Marterich, Thanks for your contribution 😄

Reply:
TLDR: I dont think the possible problems that this change could induce justify the means to use the Cmdlet over the raw application call.

First of all, Thanks for the Idea, 
I would nevertheless keep the version call raw, without using the Cmdlet. I see the point in using the Microsoft Provided functions (in case they change the inner workings of Winget later on), but it poses some difficulty as well to use it here. 
If you use the "raw" binary argument without the cmdlet, suffices if the binary is in your $PATH and can be executed by the user. 
If you however want to use the Cmdlet "Get-WinGetVersion" you have to make sure that the user has the  Module `Microsoft.WinGet.Client` installed as well before checking the version, which isn't necessarily needed for running Winget in itself. In case the Module is not installed, yet, you would then need to handle the installation, which would possibly require setting up a NuGet Provider, allowing PSAppGallery to download the module and then installing it.
So, this would require quite a lot more changes to the inner workings and error handling of winutil which seems to me a bit much to query the version. 

Nevertheless, the use of the Module `Microsoft.WinGet.Client` could be taken into consideration for for examle Listing the installed packages and so on, because it returns actual CustomPSObjects for Name, Id, Version etc. which would remidiate the need for the error prone string manipulation that is used at the moment to parse the output of the binary and possibly other tasks as well. 
However i think this would definetly need to be handled in another PR or a Discussion Thread to check if it is worth it to integrate the Module as a dependency into the project.